src/components/header/_macro.njk

```diff
@@ -1,127 +1,144 @@
-    {% set currentLanguage = params.language.languages | selectattr("current") | first %}
-    {% set currentLanguageISOCode = currentLanguage.ISOCode %}
-  {% endif %}
+    {% if params.language is defined and params.language and params.language.languages is defined and params.language.languages %}
+        {% set currentLanguage = params.language.languages | selectattr("current") | first %}
+        {% set currentLanguageISOCode = currentLanguage.ISOCode %}
+    {% endif %}
 
-  <header class="header {{ params.classes if params.classes }}" role="banner">
-      {% if params.phase is defined and params.phase %}
-          {% from "components/phase-banner/_macro.njk" import onsPhaseBanner %}
-          {{ onsPhaseBanner(params.phase) }}
-      {% endif %}
-      <div class="header__top">
-          <div class="container{{ ' container--full-width' if params.fullWidth }}{{ ' container--wide' if params.wide }}">
-              <div class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap {{ 'header__grid-top-' + params.customHeaderLogo if params.customHeaderLogo }}">
+    <header class="header {{ params.classes if params.classes }}" role="banner">
+        {% if params.phase is defined and params.phase %}
+            {% from "components/phase-banner/_macro.njk" import onsPhaseBanner %}
+            {{ onsPhaseBanner(params.phase) }}
+        {% endif %}
+        <div class="header__top">
+            <div class="container{{ ' container--full-width' if params.fullWidth }}{{ ' container--wide' if params.wide }}">
+                <div class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap {{ 'header__grid-top-' + params.customHeaderLogo if params.customHeaderLogo }}">
 
-                  <div class="grid__col col-auto">
-                    {% if params.logoHref is defined and params.logoHref %}<a class="header__logo-link" href="{{ params.logoHref }}">{% endif %}
-                      <picture>
-                        <source media="(max-width: 399px)" srcset="{{ params.assetsURL if params.assetsURL is defined and params.assetsURL }}/img/{{ params.mobileLogo | default('ons-logo-stacked-pos-' + currentLanguageISOCode) }}.svg">
-                        <img class="header__logo" src="{{ params.assetsURL if params.assetsURL is defined and params.assetsURL }}/img/{{ params.logo | default('ons-logo-pos-' + currentLanguageISOCode) }}.svg" alt="{{ params.logoAlt | default('Office for National Statistics logo') }}">
-                      </picture>
-                    {% if params.logoHref is defined and params.logoHref %}</a>{% endif %}
-                  </div>
+                    <div class="grid__col col-auto{% if params.mobileLogo is defined and params.mobileLogo %} masthead-logo-responsive{% endif %}">
+                        <div class="masthead-logo">
```

Comment:
Should it be optionally responsive? I think this should be required with defaults or if no mobile logo version is supplied then the logo should adjust as much as it can.

Also, I'm not sure about the new `masthead` class. Perhaps introduce modifiers to the existing `header__logo` i.e. `header__logo--small` & `header__logo--large` or something.